### PR TITLE
[ntuple] Improve `RNTupleInspector` documentation

### PR DESCRIPTION
**Pull request by enirolf**
This PR expands the `RNTupleInspector` documentation by adding proper Doxygen tags and adding additional information where necessary.

### CODE REVIEW

File: tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx [hunk 1]
```diff
@@ -53,15 +56,20 @@ auto file = TFile::Open("data.rntuple");
 auto rntuple = file->Get<RNTuple>("NTupleName");
 auto inspector = RNTupleInspector::Create(rntuple).Unwrap();
 
-std::cout << "The compression factor is " << std::fixed << std::setprecision(2)
-                                          << inspector->GetCompressionFactor()
-                                          << std::endl;
+std::cout << "The compression factor is " << inspector->GetCompressionFactor()
+          << " using compression settings " << inspector->GetCompressionSettings()
+          << std::endl;
```
Comment (enirolf) [inline tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx:61]: It might be worth it to have a tutorial to show more of the functionality. What do you think? This would be addressed by a separate PR>

File: tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx [hunk 2]
```diff
@@ -113,21 +126,31 @@ private:
...
-   /// Gather column-level, as well as RNTuple-level information. The column-level
-   /// information will be stored in `fColumnInfo`, and the RNTuple-level information
-   /// in `fCompressionSettings`, `fOnDiskSize` and `fInMemorySize`.
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Gather column-level and RNTuple-level information.
    ///
-   /// This method is called when the `RNTupleInspector` is initially created. This means that anything unexpected about
-   /// the RNTuple itself (e.g. inconsistent compression settings across clusters) will be detected here. Therefore, any
-   /// related exceptions will be thrown on creation of the inspector.
+   /// \note This method is called when the RNTupleInspector is initially created. This means that anything unexpected
```
Comment (vepadulano) [inline tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx:132]: nice!

File: tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx [hunk 3]
```diff
@@ -113,21 +126,31 @@ private:
...
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Recursively gather field-level information.
    ///
-   /// This method is called when the `RNTupleInspector` is initially created.
+   /// \param[in] fieldId The ID of the field from which to start the recursive traversal. Typically this is the "zero
+   /// ID", i.e. the logical parent of all top-level fields.
+   ///
+   /// \return The RFieldTreeInfo for the provided field ID.
+   ///
+   // / This method iscalled when the RNTupleInpector is initially created.
```
Comment (vepadulano) [inline tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx:145]:
```suggestion
   // / This method is called when the RNTupleInpector is initially created.
```

File: tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx [hunk 4]
```diff
@@ -137,42 +160,104 @@ public:
...
+   ///
+   /// \param[in] pageSource The RPageSource object belonging to the RNTuple to be inspected.
+   ///
+   /// \see Create(RNTuple *sourceNTuple)
+   static std::unique_ptr<RNTupleInspector> Create(std::unique_ptr<Detail::RPageSource> pageSource);
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get the descriptor for the RNTuple being inspected.
+   ///
+   /// \return A static copy of the RNTupleDescriptor belonging to the inspected RNTuple.
```
Comment (vepadulano) [inline tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx:195]: Does this return a "static copy"?

Comment (enirolf) [reply tree/ntupleutil/v7/inc/ROOT/RNTupleInspector.hxx:195]: `fDescriptor` holds a clone of the `RNTuple` that is being inspected:
https://github.com/root-project/root/blob/1a0a84806ebca04ce794db0581b054371aaac2b8/tree/ntupleutil/v7/src/RNTupleInspector.cxx#L35-L37
So it's a static copy in the sense that if for any reason the inspected `RNTuple` changes during the lifetime of the inspector object, these changes won't be reflected there. I think the word 'static' may cause confusion here though, so I'll remove it.

**Review by vepadulano — APPROVED** [review]
LGTM, thank you!
